packages/flutter/lib/src/widgets/inherited_theme.dart

```diff
@@ -6,6 +6,18 @@
 library;
 
 import 'framework.dart';
+import 'inherited_model.dart';
+
+/// A generic selector interface for theme data.
+///
+/// This interface allows for selecting specific aspects of theme data without
+/// coupling to specific theme implementations like Material or Cupertino.
+abstract interface class ThemeSelector<T, V> {
```

Comment:
What actually implements ThemeSelector and `select`? It's late on a Friday but I'm not seeing it :smile:

Reply:
This is for end user, they create selector.

Reply:
```dart
class CupertinoThemeBrightness implements ThemeSelector<CupertinoThemeData, Brightness> {
  const CupertinoThemeBrightness();

  @override
  Brightness select(CupertinoThemeData theme) => theme.brightness ?? Brightness.light;
}
```

Reply:
Updated implementation a little bit, now ThemeSelector is being used `select` method. For user, its just Simple pattern :-
```dart
final brightness = CupertinoTheme.select(context, (data) => data.brightness ?? Brightness.light);
```